File: airbyte-db/db-lib/src/main/java/io/airbyte/db/ContextQueryFunction.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.db;

import java.sql.SQLException;
import org.jooq.DSLContext;

@FunctionalInterface
public interface ContextQueryFunction<T> {

  T query(DSLContext context) throws SQLException;

}

File: airbyte-cdk/bulk/toolkits/legacy-source-integration-tests/src/testFixtures/kotlin/io/airbyte/cdk/test/fixtures/legacy/ContextQueryFunction.kt
/*
 * Copyright (c) 2026 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.test.fixtures.legacy

import java.sql.SQLException
import org.jooq.DSLContext

fun interface ContextQueryFunction<T> {
    @Throws(SQLException::class) fun query(context: DSLContext): T
}
